### ISSUE TITLE
@items is never frozen during compilation

### ISSUE THREAD

**Issue by query** [nanoc/nanoc#795]
This means that `sorted_articles` and friends inefficiently loop through the site's items no matter when they're called. I have a blog that defines a couple of helpers on top of `sorted_articles` to build archives by month and tag, and compilation is painfully slow.

This is version 4.1.1 on Ruby 2.3.0.

**Comment by ddfreyne** [comment]
This one’s a little tricky because the item *views* are never frozen explicitly—but I suppose they should mirror the frozen-ness of the backing entity. Fix coming up.

**Comment by ddfreyne** [comment]
WIP fix is in #797.

**Comment by ddfreyne** [comment]
This is now fixed by #796 in the `release-4.1.x` branch and will be part of the 4.1.2 release.

Can you verify that this fix works for you? The simplest way to use this branch is to update the `nanoc` line in your Gemfile to say the following:

```ruby
gem 'nanoc', github: 'nanoc/nanoc', branch: 'release-4.1.x'
```

**Comment by query** [comment]
Yup, this works. Compile times went down from minutes to less than 15 seconds. Thanks!
